TestBenches/MatchProcessorL3_test.cpp

```diff
@@ -0,0 +1,182 @@
+    writeMemFromFile<TrackletProjectionMemory<BARRELPS> >(tproj8, fin_tproj8, ievt);
+    writeMemFromFile<TrackletProjectionMemory<BARRELPS> >(tproj9, fin_tproj9, ievt);
+    writeMemFromFile<TrackletProjectionMemory<BARRELPS> >(tproj10, fin_tproj10, ievt);
+    writeMemFromFile<AllStubMemory<BARRELPS> >(allstub, fin_as, ievt);
+
+    writeMemFromFile<VMStubMEMemoryCM<BARRELPS, 3, 3, kNMatchEngines> >(inputvmstubs, fin_vmstub, ievt);
+
+    //set bunch crossing
+    BXType bx=ievt;
+    BXType bx_out;
+
+    int noutcandmatch = 0;
+
+    // Unit Under Test
+    MatchProcessorTopL3(bx,
+                        &tproj1, &tproj2, &tproj3, &tproj4,
+                        &tproj5, &tproj6, &tproj7, &tproj8,
+			&tproj9, &tproj10,
+                        inputvmstubs,
+                        &allstub,
+                        bx_out,
+                        fullmatch);
+
+    // compare the computed outputs with the expected ones for the candidate 
+    // matches
+    bool truncation = false;
+
+    // compare the computed outputs with the expected ones 
+    std::cout << "FM: L1L2 seeding" << std::endl;
+    err_count += compareMemWithFile<FullMatchMemory<BARREL> >(fullmatch[0], fout_fm1, ievt, "FullMatch", truncation);
+    //err_count += compareMemWithFile<FullMatchMemory<BARREL> >(fullmatch[0], fout_fm1, ievt, "FullMatch", truncation);
+    //std::cout << "FM: L3L4 seeding" << std::endl;
+    //err_count += compareMemWithFile<FullMatchMemory<BARREL> >(fullmatch2, fout_fm2, ievt, "FullMatch", truncation);
+    std::cout << "FM: L5L6 seeding" << std::endl;
+    err_count += compareMemWithFile<FullMatchMemory<BARREL> >(fullmatch[3], fout_fm3, ievt, "FullMatch", truncation);
+    //err_count += compareMemWithFile<FullMatchMemory<BARREL> >(fullmatch[2], fout_fm3, ievt, "FullMatch", truncation);
+    //std::cout << "FM: D1D2 seeding" << std::endl;
+    //err_count += compareMemWithFile<FullMatchMemory<BARREL> >(fullmatch4, fout_fm4, ievt, "FullMatch", truncation);
+    //std::cout << "FM: D3D4 seeding" << std::endl;
+    //err_count += compareMemWithFile<FullMatchMemory<BARREL> >(fullmatch5, fout_fm5, ievt, "FullMatch", truncation);
+    //std::cout << "FM: L1D1 seeding" << std::endl;
+    //err_count += compareMemWithFile<FullMatchMemory<BARREL> >(fullmatch6, fout_fm6, ievt, "FullMatch", truncation);
+    //std::cout << "FM: L2D1 seeding" << std::endl;
+    //err_count += compareMemWithFile<FullMatchMemory<BARREL> >(fullmatch7, fout_fm7, ievt, "FullMatch", truncation);
+    
+
+  }  // end of event loop
+  
+  // close files
+  fin_tproj1.close();
+  fin_tproj2.close();
+  fin_tproj3.close();
+  fin_tproj4.close();
+  fin_tproj5.close();
+  fin_tproj6.close();
+  fin_tproj7.close();
+  fin_tproj8.close();
+  fin_vmstub.close();
+
+  fout_fm1.close();
+  fout_fm2.close();
+  fout_fm3.close();
+  fout_fm4.close();
+  fout_fm5.close();
+  fout_fm6.close();
+  fout_fm7.close();
+
+  return err_count;
```

Comment:
All the other modules has an extra if statement here to avoid the testbench to false pass. I guess it's good to add it here as well? See: https://github.com/cms-L1TK/firmware-hls/blob/MPtest/TestBenches/MatchEngine_test.cpp#L106-L108

Reply:
Thanks, I've added that.